src/assets/scripts/client/airport/MapLayerModel.js

```diff
@@ -0,0 +1,118 @@
+import _forEach from 'lodash/forEach';
+import _isEqual from 'lodash/isEqual';
+import _isNumber from 'lodash/isNumber';
+import _map from 'lodash/map';
+import BaseModel from '../base/BaseModel';
+import DynamicPositionModel from '../base/DynamicPositionModel';
+import StaticPositionModel from '../base/StaticPositionModel';
+
+/**
+ * A collection of lines that represent a map layer.
+ *
+ * @class MapLayerModel
+ */
+export default class MapLayerModel extends BaseModel {
+    /**
+     * @for MapLayerModel
+     * @constructor
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(map, airportPosition, magneticNorth) {
+        super();
+
+        if (!map || !airportPosition || !_isNumber(magneticNorth)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid parameter, expected map, airportPosition and magneticNorth to be defined');
+        }
+
+        /**
+         * A flag indicating whether the layer should be hidden
+         *
+         * @for MapLayerModel
+         * @property hidden
+         * @type {boolean}
+         * @default false
+         */
+        this.hidden = false;
+
+        /**
+         * List of lat/long coordinates pairs that define a line
+         *
+         * @for MapLayerModel
+         * @property lines
+         * @type {array}
+         * @default []
+         */
+        this.lines = [];
+
+        /**
+         * Name of the map layer.
+         *
+         * @for MapLayerModel
+         * @property name
+         * @type {string}
+         * @default ''
+         */
+        this.name = '';
+
+        return this._init(map, airportPosition, magneticNorth);
+    }
+
+    /**
+     * Initialize the model
+     *
+     * @for MapLayerModel
+     * @method _init
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     * @private
+     */
+    _init(map, airportPosition, magneticNorth) {
+        this.hidden = map.hidden === true;
+        this.name = map.name;
+        this._buildMapLines(map.lines, airportPosition, magneticNorth);
```

Comment:
missing empty line between assignments and method call

Reply:
Again, I didn't know about that convention - I copied the style from AirspaceModel where there's no empty line.

I guess you mean this?
```js
    _init(map, airportPosition, magneticNorth) {
        this.isHidden = map.hidden === true;
        this.name = map.name;
        
        this._buildMapLines(map.lines, airportPosition, magneticNorth);

        return this;
    }

```

Reply:
Correct. Any time you change between assignments, method/function calls, or returns, there should be a blank line. Otherwise, no blank lines. So your example is correct.